### ISSUE TITLE
Bug: no-unused-vars does not work in default import 

### ISSUE THREAD

**Issue by Rayzhangzhang** [typescript-eslint/typescript-eslint#5816]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

-

### Repro Code

```typescript
import a from './a'

'a' is declared but its value is never read.ts(6133), ts works well but typescript-eslint doesn't work.
```

### ESLint Config

```javascript
'no-unused-vars': 'off',
'@typescript-eslint/no-unused-vars': [1, { argsIgnorePattern: '^_', varsIgnorePattern: '^ignored?$' }],
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "noImplicitReturns": true,
    "suppressImplicitAnyIndexErrors": true,
    "noUnusedLocals": true,
  }
}
```

### Expected Result

works.

### Actual Result

doesn't work

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `TypeScript`                           | `4.8.4` |
| `ESLint`                               | `8.25.0` |
| `node`                                 | `16.14.0` |

**Comment by bradzacher** [comment]
please include a playground repro for your bug report

**Comment by JoshuaKGoldberg** [comment]
If anybody wants to file a new issue that adheres to the template (and so provides enough info for us to action on it), please do! 🙂
